user update password, upload profile image, update address function, and bug fixes

alicenguyen2529 opened the pull request.

SamuelHobbsHCL reviewed (commented):
- Please delegate the logic in UserController to UserService; I recommend returning something for the delete function rather than void (to signify if anything went wrong).
- Don't set a default address for registerUser since we should delegate this in another ticket - also, please disable the site's showing the "update contact info" until we have it in for real
- I noticed some of these open new tabs - I'd recommend against it for consistency w/the rest of our site's design

alicenguyen2529 commented:
> * Please delegate the logic in UserController to UserService; I recommend returning something for the delete function rather than void (to signify if anything went wrong).
> * Don't set a default address for registerUser since we should delegate this in another ticket - also, please disable the site's showing the "update contact info" until we have it in for real
> * I noticed some of these open new tabs - I'd recommend against it for consistency w/the rest of our site's design

@SamuelHobbsHCL Thank you for your feed back! I updated the UserController per feedback 1 and update the UI to not open new tabs per feedback 2. For feedback 2, I have added the user update address function. The flow is: when user register they don't have to put in address (this is similar to a real world experience). They will be assigned a default address. They can go in their user profile page to update to their address (The user will not be able to checkout unless they have updated address).

Gek0r reviewed (approved):
Looks good on my end, nice work!
